scm/github.go

```diff
@@ -689,6 +689,108 @@ func (s *GithubSCM) GetUserScopes(ctx context.Context) *Authorization {
 	return &Authorization{Scopes: gitScopes}
 }
 
+// CreateIssue implements the SCM interface
+func (s *GithubSCM) CreateIssue(ctx context.Context, opt *CreateIssueOptions) (*Issue, error) {
+	if !opt.valid() {
+		return nil, ErrMissingFields{
+			Method:  "CreateIssue",
+			Message: fmt.Sprintf("%+v", opt),
+		}
+	}
+	newIssue := &github.IssueRequest{
+		Title:     &opt.Title,
+		Body:      &opt.Body,
+		Assignee:  opt.Assignee,
+		Assignees: opt.Assignees,
+	}
+
+	s.logger.Debugf("CreateRepository: creating issue %s on %s Repository", opt.Title, opt.Repository)
+	issue, _, err := s.client.Issues.Create(ctx, opt.Organization, opt.Repository, newIssue)
+	if err != nil {
+		return nil, ErrFailedSCM{
+			Method:   "CreateIssue",
+			Message:  fmt.Sprintf("failed to create Issue %s, make sure all the details are correct ", opt.Title),
+			GitError: err,
+		}
+	}
+	s.logger.Debugf("CreateRepository: done creating issue %s", opt.Title)
+
+	return toIssue(issue), nil
+}
+
+// GetRepoIssue implements the SCM interface
+func (s *GithubSCM) GetRepoIssue(ctx context.Context, opt *IssueOptions) (*Issue, error) {
```

Comment:
`IssueOptions` looks very similar to `RepositoryOptions`.

You could do this:
```go
func (s *GithubSCM) GetRepoIssue(ctx context.Context, issueNumber int, opt *RepositoryOptions) (*Issue, error) {
	if !opt.valid() { // Valid as long as either opt.ID > 0 || (opt.Path != "" && opt.Owner != "")
		...
	}
	issue, _, err := s.client.Issues.Get(ctx, opt.Owner, opt.Path, issueNumber)
```

And similarly for `GetRepoIssues`, but omitting `issueNumber int`

Reply:
Doing this, and as mentioned below, you can remove `IssueOptions` altogether from `scm.go`.

Reply:
This also requires some small adjustments in the tests.